### PR TITLE
feat: switch to Github managed M1 runners that are in beta now

### PR DESCRIPTION
**Pull request by ansgarm**
Related to https://github.com/actions/runner-images/issues/8439

### CODE REVIEW

**Comment by ansgarm** [comment]
Here's a dry-run that I triggered for this branch: https://github.com/cdktf/node-pty-prebuilt-multiarch/actions/runs/6429875324

**Review by mutahhir — APPROVED** [review]

File: .prebuild/Dockerfile.debian [hunk 1]
```diff
@@ -1,13 +1,13 @@
 ARG BASE_IMAGE
-FROM ${BASE_IMAGE:-library/debian:9.6-slim}
```
Comment (shahardra) [inline .prebuild/Dockerfile.debian:2]: .github/workflows/release.yml

File: .prebuild/Dockerfile.debian [hunk 2]
```diff
@@ -1,13 +1,13 @@
...
+FROM ${BASE_IMAGE:-library/debian:11.7-slim}
 
 ARG QEMU_ARCH
 ENV QEMU_ARCH=${QEMU_ARCH:-x86_64}
 
 COPY .prebuild/qemu/qemu-${QEMU_ARCH}-static /usr/bin/
 
 RUN apt-get update \
-  && apt-get install -y gcc g++ make python curl
```
Comment (shahardra) [inline .prebuild/Dockerfile.debian:10]: .github/workflows/release.yml

File: .prebuild/Dockerfile.debian (hunk 1 above)
Comment (shahardra) [inline .prebuild/Dockerfile.debian:2]: .github/workflows/release.yml

File: .prebuild/Dockerfile.debian (hunk 2 above)
Comment (shahardra) [inline .prebuild/Dockerfile.debian:10]: .github/workflows/release.yml

File: .prebuild/qemu/README.md [hunk 3]
```diff
@@ -0,0 +1,5 @@
+Source: https://github.com/NorthernMan54/node-pty-prebuilt-multiarch/blob/401d14756196d24dabeadc7b55cff28dfea68d4d/.prebuild/qemu/README.md
+
+These are copied from https://github.com/multiarch/qemu-user-static/releases/
+
+Current release - qemu-*-static @ 7.2.0-1
```
Comment (shahardra) [inline .prebuild/qemu/README.md:5]: .github/workflows/release.yml

File: build.sh [hunk 4]
```diff
@@ -8,19 +8,19 @@ export DOCKERFILE="Dockerfile.debian"
 docker build -f .prebuild/$DOCKERFILE --build-arg BASE_IMAGE=${BASE_IMAGE} --build-arg QEMU_ARCH=${QEMU_ARCH} -t multiarch-build .
 docker run --rm -v $(pwd):/node-pty multiarch-build
 
-export BASE_IMAGE=library/debian:9.6-slim
```
Comment (shahardra) [inline build.sh:11]: .github/workflows/release.yml

File: build.sh [hunk 5]
```diff
@@ -8,19 +8,19 @@ export DOCKERFILE="Dockerfile.debian"
...
+export BASE_IMAGE=library/debian:11.7-slim
 export QEMU_ARCH=x86_64
 export DOCKERFILE="Dockerfile.debian"
 docker build -f .prebuild/$DOCKERFILE --build-arg BASE_IMAGE=${BASE_IMAGE} --build-arg QEMU_ARCH=${QEMU_ARCH} -t multiarch-build .
 docker run --rm -v $(pwd):/node-pty multiarch-build
 
-export BASE_IMAGE=i386/debian:9.6-slim
```
Comment (shahardra) [inline build.sh:17]: .github/workflows/release.yml

File: build.sh [hunk 6]
```diff
@@ -8,19 +8,19 @@ export DOCKERFILE="Dockerfile.debian"
...
+export BASE_IMAGE=i386/debian:11.7-slim
 export QEMU_ARCH=i386
 export DOCKERFILE="Dockerfile.debian"
```
Comment (shahardra) [inline build.sh:19]: .github/workflows/release.yml

File: build.sh [hunk 7]
```diff
@@ -8,19 +8,19 @@ export DOCKERFILE="Dockerfile.debian"
...
-export BASE_IMAGE=library/debian:9.6-slim
+export BASE_IMAGE=library/debian:11.7-slim
 export QEMU_ARCH=x86_64
 export DOCKERFILE="Dockerfile.debian"
 docker build -f .prebuild/$DOCKERFILE --build-arg BASE_IMAGE=${BASE_IMAGE} --build-arg QEMU_ARCH=${QEMU_ARCH} -t multiarch-build .
 docker run --rm -v $(pwd):/node-pty multiarch-build
 
-export BASE_IMAGE=i386/debian:9.6-slim
+export BASE_IMAGE=i386/debian:11.7-slim
 export QEMU_ARCH=i386
```
Comment (shahardra) [inline build.sh:18]: .github/workflows/release.yml

File: build.sh [hunk 8]
```diff
@@ -8,19 +8,19 @@ export DOCKERFILE="Dockerfile.debian"
...
 export DOCKERFILE="Dockerfile.debian"
 docker build -f .prebuild/$DOCKERFILE --build-arg BASE_IMAGE=${BASE_IMAGE} --build-arg QEMU_ARCH=${QEMU_ARCH} -t multiarch-build .
 docker run --rm -v $(pwd):/node-pty multiarch-build
 
-export BASE_IMAGE=arm64v8/debian:9.6-slim
```
Comment (shahardra) [inline build.sh:23]: .github/workflows/release.yml

File: build.sh [hunk 9]
```diff
@@ -8,19 +8,19 @@ export DOCKERFILE="Dockerfile.debian"
...
+export BASE_IMAGE=arm64v8/debian:11.7-slim
 export QEMU_ARCH=aarch64
```
Comment (shahardra) [inline build.sh:24]: .github/workflows/release.yml

File: build.sh [hunk 10]
```diff
@@ -8,19 +8,19 @@ export DOCKERFILE="Dockerfile.debian"
...
 export DOCKERFILE="Dockerfile.debian"
```
Comment (shahardra) [inline build.sh:25]: .github/workflows/release.yml

**Review by shahardra — COMMENTED** [review]
.github/workflows/release.yml

**Comment by radeva** [comment]
@ansgarm @mutahhir In case you need some free usage of M1 runners, you can try [FlyCI](https://www.flyci.net/)'s M1 and M2 runners. They are on average 2x faster and 2x cheaper than GitHub's **AND** we have a **free tier for OSS projects** (see below).

# Install Instructions

1. Install [FlyCI app](https://github.com/apps/flyci-prod) and 
2. Easily replace one line of code and start using FlyCI runners:

```diff
jobs:
 ci:
-    runs-on: macos-latest
+    runs-on: flyci-macos-large-latest-m1
   steps:
   - name: 👀 Checkout repo
     uses: actions/checkout@v4
```
# 500 mins/month Free for Public Repos

Since your repo is public, FlyCI offers **500 mins/month of free M1 runner usage** with the `flyci-macos-large-latest-m1` runner for public projects. 

Don't hesitate to contact us in case the free tier doesn't suit your needs or you experience any issues with the runners. Our team is here to support you!

Best Regards,
Veselina Radeva
Product Manager at [FlyCI](https://flyci.net/)

**Pull request merged**